alpha/lib/model/LiveEntry.php

```diff
@@ -499,41 +499,133 @@ public function getScheduleEvents($startTime, $endTime)
 		return $events;
 	}
 
+	public function getLiveStatus($checkExplicitLive = false, $protocol = null)
+	{
+		if ($checkExplicitLive && $this->getViewMode() == ViewMode::PREVIEW && !$this->canViewExplicitLive())
+		{
+			return EntryServerNodeStatus::STOPPED;
+		}
+
+		if (in_array($this->getSource(), LiveEntry::$kalturaLiveSourceTypes))
+		{
+			return $this->getInternalLiveStatus($checkExplicitLive);
+		}
+		else
+		{
+			// for external entry we have only live or offline status
+			return $this->isExternalCurrentlyLive($protocol) ? EntryServerNodeStatus::PLAYABLE : EntryServerNodeStatus::STOPPED;
+		}
+	}
 
 	/**
 	 * @return boolean
 	 */
-	public function isCurrentlyLive($currentDcOnly = false)
+	public function isCurrentlyLive($currentDcOnly = false, $protocol = null)
+	{
+		return $this->getLiveStatus(true, $protocol) === EntryServerNodeStatus::PLAYABLE;
+	}
+
+	private function isExternalCurrentlyLive($reqProtocol = null)
 	{
-		if ($this->getViewMode() == ViewMode::PREVIEW)
+		$isLive = null;
+		$takeFirst = false;
+		$protocols = array();
+		switch ($reqProtocol)
 		{
-			if (!$this->canViewExplicitLive())
-				return false;
+			case KalturaPlaybackProtocol::HLS:
+			case KalturaPlaybackProtocol::APPLE_HTTP:
+				$protocols = array_unique(array($reqProtocol, KalturaPlaybackProtocol::HLS, KalturaPlaybackProtocol::APPLE_HTTP));
+				$takeFirst = true;
+				break;
+			case KalturaPlaybackProtocol::HDS:
+			case KalturaPlaybackProtocol::AKAMAI_HDS:
+				$protocols = array($reqProtocol);
+				break;
+
+			case null:
+				$configurations = $this->getLiveStreamConfigurations(requestUtils::getProtocol());
+				foreach ($configurations as $config)
+				{
+					$protocols[] = $config->getProtocol();
+				}
+				break;
 		}
 
-		if (kSimuliveUtils::getPlayableSimuliveEvent($this))
+		foreach($protocols as $protocol)
 		{
-			return true;
+			$isProtocolLive = $this->checkIsLiveByProtocol($protocol);
+			if ($isProtocolLive !== null)
+			{
+				if ($isProtocolLive || $takeFirst)
+				{
+					return $isProtocolLive;
+				}
+				$isLive = $isProtocolLive; // as false
```

Comment:
should we break the loop once $isLive is set?

Reply:
No - as is isLive is false by one protocol we want to keep check all other optional protocol (that the behavior we had till now)